openapi.yaml

```diff
@@ -0,0 +1,787 @@
+    post:
+      operationId: createPermissionSet
+      tags:
+        - "permissionsets"
+      description: Create a new Permission Set
+      requestBody:
+        content:
+          'application/json':
+            schema:
+              $ref: "#/components/schemas/PermissionSetRequest"
+        required: true
+      responses:
+        "201":
+          description: "Display created PermissionSet"
+          content:
+            "application/json":
+              schema:
+                $ref: "#/components/schemas/PermissionSetResponse"
+        "400":
+          $ref: "#/components/responses/BadInput"
+        "500":
+          $ref: "#/components/responses/ServerError"
+  /permission-sets/{permissionSetId}:
+    parameters:
+      - name: "permissionSetId"
+        in: "path"
+        description: "Id of PermissionSet to return"
+        required: true
+        schema:
+          type: "string"
+    get:
+      operationId: readPermissionSet
+      description: Access an existing Permission Set
+      tags:
+        - "permissionsets"
+      responses:
+        "200":
+          description: "Display Permission Set"
+          content:
+            "application/json":
+              schema:
+                $ref: "#/components/schemas/PermissionSetResponse"
+        "404":
+          $ref: "#/components/responses/NotFound"
+        "500":
+          $ref: "#/components/responses/ServerError"
+    put:
+      operationId: updatePermissionSet
+      description: Update an existing Permission Set
+      tags:
+        - "permissionsets"
+      requestBody:
+        required: true
+        content:
+          'application/json':
+            schema:
+              $ref: "#/components/schemas/PermissionSetRequest"
+      responses:
+        "200":
+          description: "Display updated Permission Set"
+          content:
+            "application/json":
+              schema:
+                $ref: "#/components/schemas/PermissionSetResponse"
+        "400":
+          $ref: "#/components/responses/BadInput"
+        "404":
+          $ref: "#/components/responses/NotFound"
+        "500":
+          $ref: "#/components/responses/ServerError"
+    delete:
+      operationId: deletePermissionSet
+      description: Delete an existing Permission Set
+      tags:
+        - "permissionsets"
+      responses:
+        "200":
+          $ref: "#/components/responses/Deleted"
+        "404":
+          $ref: "#/components/responses/NotFound"
+        "500":
+          $ref: "#/components/responses/ServerError"
+  /permissions:
+    get:
+      operationId: getPermissions
+      tags:
+        - "permissionsets"
+      responses:
+        "200":
+          description: "Display available permissions"
+          content:
+            "application/json":
+              schema:
+                $ref: "#/components/schemas/PermissionsResponse"
+  /grants:
+    get:
+      operationId: listGrants
+      tags:
+        - "grants"
+      parameters:
```

Comment:
How do we list all grants assigned to a user?  I only see query parameters for filtering grants by securable or by permission set.  This is a P1 requirement, see https://docs.google.com/document/d/1EtqvrG98fvSKjSsm5UMbWt3ejVacGaUtn3dnx88mt18/edit?disco=AAAAbj1fu24